src/components/pages/Publish/Debug.tsx

```diff
@@ -9,25 +9,30 @@ export default function Debug({
 }: {
   values: Partial<MetadataPublishFormDataset>
 }): ReactElement {
-  const ddo = {
-    '@context': 'https://w3id.org/did/v1',
-    dataTokenInfo: {
-      ...values.dataTokenOptions
-    },
-    service: [
-      {
-        index: 0,
-        type: 'metadata',
-        attributes: { ...transformPublishFormToMetadata(values) }
-      },
-      {
-        index: 1,
-        type: values.access,
-        serviceEndpoint: values.providerUri,
-        attributes: {}
-      }
-    ]
-  }
+  const [ddo, setDdo] = useState({})
```

Comment:
What is the reason for this change?

Reply:
Following the existing pattern of the filters, `transformPublishFormToMetadata` is now asynchronous because I had to check whether the SD is in compliance and set the `isInCompliance` property  before I publish the asset. The compliance verification is async and happens inside of `transformPublishFormToMetadata` so I had to update every call to it.